crates/swc_ecma_minifier/src/analyzer/storage/normal.rs

```diff
@@ -85,9 +85,9 @@ impl Storage for ProgramData {
 
                     e.get_mut().pure_fn |= var_info.pure_fn;
 
-                    if !var_info.is_fn_local {
-                        e.get_mut().is_fn_local = false;
-                    }
+                    e.get_mut().used_recursively |= var_info.used_recursively;
+
+                    e.get_mut().is_fn_local &= var_info.is_fn_local;
```

Comment:
I think we need `e.get_mut().used_in_non_child_fn &= var_info.used_in_non_child_fn;`